src/poetry/utils/env/env_manager.py

```diff
@@ -441,10 +442,20 @@ def create_venv(
             venv = venv_path / name
 
         if venv_prompt is not None:
-            venv_prompt = venv_prompt.format(
-                project_name=self._poetry.package.name or "virtualenv",
-                python_version=python.minor_version.to_string(),
-            )
+            try:
+                venv_prompt = venv_prompt.format(
+                    project_name=self._poetry.package.name or "virtualenv",
+                    python_version=python.minor_version.to_string(),
+                )
+            except KeyError as e:
+                raise PoetryError(
+                    f'Invalid template variable "{e.args[0]}" in '
+                    'global configuration option "virtualenvs.prompt".\n\n'
```

Comment:
It is not necessarily in the global configuration. It can also be in the [local configuration](https://python-poetry.org/docs/configuration/#local-configuration).